### PR TITLE
REGD-170-nearby3

### PR DESCRIPTION
**Pull request by mingjiecn**
There are several changes needed after the regulome meeting discussion:
1. Variants in LD are still to far away to be shown in the same track with the SNP got hit, so we need to use a different scale to show them in a different track
2. We will only show regulatory region from ccre for now instead of all sources.
3. We will also show the nearest SNPs in the same track with the SNP got hit.
4. Add some space on both side of the view box so the label of the scale will not be cut off.
5. Update fetch_variant_ld to sort the items in return so it is easy to use in nearby drawing.

### CODE REVIEW

**Review by forresttanaka — APPROVED** [review]

File: components/nearby-diagram.js [hunk 1]
```diff
@@ -201,35 +201,46 @@ export default function NearbyDiagram({
...
+        zoomInIconPositionForVariantsInLd + zoomInIconHeigt + 30 + blankHeight;
+      VariantsInLdPositionY = scaleForVariantsInLdPositionY + blankHeight;
+    } else {
+      scaleForVariantInLd = scaleForGene;
+      offsetForVariantInLd = offsetXForGene;
+      VariantsInLdPositionY =
+        regRegionPositionY +
+        Object.keys(regulatoryRegionsBySource).length *
+          (regRegionPositiontrackHeight + labelHeight);
+    }
```
Comment (forresttanaka) [inline components/nearby-diagram.js:243]: Sort of odd to have an upper-case variable that’s not a React component. Any reason it gets special treatment?

File: components/nearby-diagram.js [hunk 2]
```diff
@@ -261,20 +271,28 @@ export default function NearbyDiagram({
...
     let rectY = highestLabelY;
-    if (preRectX2 >= rectX) {
-      rectY = preRectY + labelHeight + blankHeight;
+    if (preLabelInfo) {
+      const pre = preLabelInfo.find((item) => item < rectX);
+      if (pre) {
+        indexY = preLabelInfo.indexOf(pre);
+      } else {
+        indexY = preLabelInfo.length;
+      }
```
Comment (forresttanaka) [inline components/nearby-diagram.js]: Conditional expression would shorten this a bit, though functionally the same.

**Review by forresttanaka — APPROVED** [review]
Just a couple minor things.

File: components/nearby-diagram.js (hunk 1 above)
Comment (mingjiecn) [reply components/nearby-diagram.js:243]: I will rename it. Thank for spotting it!

File: components/nearby-diagram.js (hunk 2 above)
Comment (mingjiecn) [reply components/nearby-diagram.js]: Will shorten it. Thanks!

**Pull request merged**